test/e2e/network/service.go

```diff
@@ -1166,10 +1205,9 @@ var _ = SIGDescribe("Services", func() {
 	ginkgo.It("should work after restarting apiserver [Disruptive]", func() {
 		// TODO: use the ServiceTestJig here
 		e2eskipper.SkipUnlessProviderIs("gce", "gke")
```

Comment:
is this a gce test only?

Reply:
yes, same as test case ("should work after restarting kube-proxy [Disruptive]").  I try to remove this skip, but no success. Beacause I find that  next the test restart api-server or kube-proxy  will still judge  the env is gce or gke. 

https://github.com/kubernetes/kubernetes/blob/fcba5fe4e4eb1b1a2e2b14d2a68037fecd75e3ab/test/e2e/network/service.go#L712-L716

and

https://github.com/kubernetes/kubernetes/blob/fcba5fe4e4eb1b1a2e2b14d2a68037fecd75e3ab/test/e2e/network/service.go#L3802-L3807

Reply:
and what happens if you remove the check for `ProviderIs` in those functions? does it work? can you add a temp commit to this PR removing it to see the result?
I mean, the test is tagged as disruptive, people can filter based on the tags, and the idea is to remove all the provider specific code ...

Reply:
All right. I think these two test case can be discussed about. Before I say that that no success, it was running in my local robot using local-cluster-up.sh to simulate a k8s cluster. But when I see the code,  I find that the code treat kube-apiserver and kube-proxy as container pod which is different from the process in my local env. So the test will failed in my env definitely.